DFC.ServiceTaxonomy.Theme/wwwroot/Scripts/trumbowyg.gds.js

```diff
@@ -261,6 +276,66 @@
         trumbowyg.restoreRange();
     }
 
+    function youtubeLink(trumbowyg) {
+
+        var options = {
+            url: {
+                label: trumbowyg.lang.iframeSettings.url,
+                required: true
+            },
+            title: {
+                label: trumbowyg.lang.iframeSettings.title
+            },
+            height: {
+                label: trumbowyg.lang.iframeSettings.height,
+                required: true,
+                value: "360"
+            },
+            width: {
+                label: trumbowyg.lang.iframeSettings.width,
+                required: true,
+                value: "640"
+            },
+        };
+
+        trumbowyg.openModalInsert(
+            trumbowyg.lang.youtubeLink,
+            options,
+            function (data) {
+
+                var srcUrl = insertQueryParamString(data.url,"rel", "?")
+
+                var $linkToEmbed = $('<iframe>')
+                    .attr('src', srcUrl)
+                    .attr('height', data.height)
+                    .attr('width', data.width)
+                    .attr('frameborder', "0")
+
+                if (data.title.length > 0)
+                    $linkToEmbed.attr('title', data.title + ' (video)')
+
+                trumbowyg.execCmd('insertHTML', $linkToEmbed.get(0).outerHTML)
+
+                return true;
+            });
+    }
+
+    function insertQueryParamString(url, queryParam, at) {
```

Comment:
This method uses both `queryParam` (with a hard-coded value) and also hard-codes the param in the function. It would be better if it did one or the other.

Also, it might be better to use [inbuilt functionality](https://stackoverflow.com/questions/486896/adding-a-parameter-to-the-url-with-javascript) to add the query param, rather than rolling our own code.

Reply:
Update code to use inbuilt url function: 'url.searchParams.set' to set query params.